### PR TITLE
Ratvar summoning now ends the round/triggers mass proselytize 

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
Mass Proselytize (the giant spiral of turf conversion) was always my favourite thing about old clock cult so I made it trigger when Ratvar arrives.

Ratvar arriving now ends the round because 90 seconds of postgame scarab party is preferable in my mind to 6 and half minutes of it

Ratvar act on mobs now just sets them on fire rather than trying to convert them. I'm not really attached to this change, I just think its weird to see all the people who lost the siege in the list of cultists/taking the same victory lap

### CODE REVIEW

File: code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm [hunk 1]
```diff
@@ -242,6 +242,20 @@
 				var/obj/structure/destructible/clockwork/massive/ratvar/R = new(startpoint)
 				var/turf/T =  locate(round(world.maxx * 0.5, 1), round(world.maxy * 0.5, 1), ZLEVEL_STATION_PRIMARY) //approximate center of the station
 				R.forceMove(T)
+				SSticker.force_ending = TRUE
+				var/x0 = T.x
+				var/y0 = T.y
+				for(var/I in spiral_range_turfs(255, startpoint))
```
Comment (Cyberboss) [inline code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm]: use the tick_checked parameter

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm [hunk 2]
```diff
@@ -242,6 +242,19 @@
 				var/obj/structure/destructible/clockwork/massive/ratvar/R = new(startpoint)
 				var/turf/T =  locate(round(world.maxx * 0.5, 1), round(world.maxy * 0.5, 1), ZLEVEL_STATION_PRIMARY) //approximate center of the station
 				R.forceMove(T)
+				SSticker.force_ending = TRUE
+				var/x0 = T.x
+				var/y0 = T.y
+				for(var/I in spiral_range_turfs(255, startpoint, tick_checked = TRUE))
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm]: should be T rather than startpoint.

**Comment by Xhuis** [comment]
Eh, why not? Works for me

File: code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm (hunk 2 above)
Comment (Cyberboss) [reply code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm]: why create him at startpoint then forceMove him to T though?

File: code/modules/mob/living/living_defense.dm [hunk 3]
```diff
@@ -338,18 +338,13 @@
...
-			adjustFireLoss(35)
-			if(src)
-				adjust_fire_stacks(1)
-				IgniteMob()
-			return FALSE
-	return TRUE
+		to_chat(src, "<span class='userdanger'>A blinding light boils you alive! <i>Run!</i></span>")
+		adjustFireLoss(35)
+		if(src)
+			adjust_fire_stacks(1)
```
Comment (Cyberboss) [inline code/modules/mob/living/living_defense.dm]: Do at least 4 fire stacks for the big lightup

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/living_defense.dm [hunk 4]
```diff
@@ -340,9 +340,8 @@
 		return
 	if(stat != DEAD && !is_servant_of_ratvar(src))
 		to_chat(src, "<span class='userdanger'>A blinding light boils you alive! <i>Run!</i></span>")
-		adjustFireLoss(35)
 		if(src)
```
Comment (Cyberboss) [inline code/modules/mob/living/living_defense.dm]: This if is stupid

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/living_defense.dm (hunk 4 above)
Comment (Xhuis) [reply code/modules/mob/living/living_defense.dm]: Some mobs delete on death, so hitting them with fireloss might kill them, wouldn't it?

**Comment by AnonymousNow** [comment]
Invading Centcom as a victorious cult with infinite power against adminspawned resistance is greatly enjoyable.

File: code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm [hunk 5]
```diff
@@ -235,13 +235,24 @@
...
+				for(var/I in spiral_range_turfs(255, T, tick_checked = TRUE))
+					var/turf/T2 = I
+					if(!T2)
+						continue
+					var/dist = cheap_hypotenuse(T2.x, T2.y, x0, y0)
+					if(dist < 100)
+						dist = TRUE
+					else
+						dist = FALSE
+					T.ratvar_act(dist)
```
Comment (Cyberboss) [inline code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm]: This still needs to be CHECK_TICK'd

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by DoctoreMad** [comment]
>I just think its weird to see all the people who lost the siege in the list of cultists/taking the same victory lap

For the kingdom of _Rat'Var_ is like a landowner who went out early in the morning to hire workers for his vineyard
- - - - - - - -
“He answered one of them, ‘I am not being unfair to you, friend. Didn’t you agree to work for a denarius?  Take your pay and go. I want to give the one who was hired last the same as I gave you. Don’t I have the right to do what I want with my own money? Or are you envious because I am generous?’

**Comment by AuroranAI** [comment]
Wouldn't this make ratvar always when when a god vs god scenario happens, since who wins that fight is determined by followers?

**Comment by Bawhoppen** [comment]
@AuroranAI, No, Why would it?

**Comment by Cyberboss** [comment]
@KorPhaeron

**Comment by KorPhaeron** [comment]
:tada:

**Pull request merged**